creds/netrc.go

```diff
@@ -0,0 +1,130 @@
+package creds
+
+import (
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/git-lfs/git-lfs/config"
+	"github.com/git-lfs/go-netrc/netrc"
+	"github.com/rubyist/tracerx"
+)
+
+type NetrcFinder interface {
+	FindMachine(string) *netrc.Machine
+}
+
+func ParseNetrc(osEnv config.Environment) (NetrcFinder, string, error) {
+	home, _ := osEnv.Get("HOME")
+	if len(home) == 0 {
+		return &noFinder{}, "", nil
+	}
+
+	nrcfilename := filepath.Join(home, netrcBasename)
+	if _, err := os.Stat(nrcfilename); err != nil {
+		return &noFinder{}, nrcfilename, nil
+	}
+
+	f, err := netrc.ParseFile(nrcfilename)
+	return f, nrcfilename, err
+}
+
+type noFinder struct{}
+
+func (f *noFinder) FindMachine(host string) *netrc.Machine {
+	return nil
+}
+
+// NetrcCredentialHelper retrieves credentials from a .netrc file
+type netrcCredentialHelper struct {
+	netrcFinder NetrcFinder
+	skip        map[string]bool
+}
```

Comment:
I really like the approach of structuring this as a credential helper.  It makes the entire authentication path much easier to reason about.

Reply:
Thanks, it was actually @ttaylorr's suggestion originally!

I like the way it turned out, I agree it makes authentication much easier to reason about. Another big plus: `.netrc` credentials were previously wrongly taking priority over credentials provided directly in a URL, and now they won't since credential helpers are queried at the end of the auth flow!